contrib/mesos/pkg/minion/tasks/task.go

```diff
@@ -240,7 +247,8 @@ type exitError interface {
 	Sys() interface{}
 }
 
-func taskRunning(t *Task) taskStateFn {
+func taskRunning(t *Task) (nextState taskStateFn) {
```

Comment:
I'm actually pretty happy with the last commit I pushed. cleans up
taskRunning a lot.

The for-loop doesn't really create a nested state machine. I could unroll
the loop. That just creates more code. We need to wait for multiple
signals: task-death, should-quit. Executing the cmd.Wait() in a goroutine
seems pretty natural and allows us to wait for multiple signals. Adding
another state machine state doesn't make sense to be because we're still
waiting for running task to stop. task-running-should-die-{soon,now} as
states seem superfluous IMO.


On Thu, Sep 24, 2015 at 10:45 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/minion/tasks/task.go
> <https://github.com/kubernetes/kubernetes/pull/14489#discussion_r40326735>
> :
>
> > @@ -240,7 +247,8 @@ type exitError interface {
> >  	Sys() interface{}
> >  }
> >
> > -func taskRunning(t *Task) taskStateFn {
> > +func taskRunning(t *Task) (nextState taskStateFn) {
>
> You can parametrize the state fn. then the code exists only once. But the
> state machine via the for-loop inside a state machine is superfluous IMO.
> Just create a taskTerminating with force argument.
> … <#14fffd16b7d8b3a4_>
> Am 24.09.2015 um 16:27 schrieb James DeFelice <notifications@github.com>:
> In contrib/mesos/pkg/minion/tasks/task.go: > @@ -240,7 +247,8 @@ type
> exitError interface { > Sys() interface{} > } > > -func taskRunning(t
> *Task) taskStateFn { > +func taskRunning(t *Task) (nextState taskStateFn) {
> This seemed cleaner, more DRY. Otherwise basically the same code blob gets
> copy-pasted elsewhere. I'm happy to refactor the loop into a separate func
> that's more easily testable. — Reply to this email directly or view it on
> GitHub.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14489/files#r40326735>.
>